machine/init.sls

```diff
@@ -0,0 +1,56 @@
+# This state installs the Gitlab-maintained fork of docker-machine
+
+{% set version = salt['pillar.get']('docker-machine:version') %}
+
+include:
+  - docker
+
+/usr/local/bin/docker-machine:
```

Comment:
How are updates handled? If I see correctly, this wouldn't update the binary if `docker-machine:version` is changed, would it?

Reply:
I was under the impression, that a `file.managed` would be updated, if the `source` field changed, at least when reapplying the state. I am not sure if changing pillar data on the `master` would trigger a reevaluation of the minion states or if that generally needs to be triggered manually.

How do we usually handle these kind of situations?

Side-Note: simply replacing the binary should be sufficient for an update, as there are no service attached.

Reply:
Ah right, with `file.managed` that's the case. Most things are `tar.gz` or similar images, which require `archive.extract` instead, my bad confused this.